qualtran/bloqs/basic_gates/global_phase.py

```diff
@@ -49,6 +64,33 @@ def decompose_bloq(self) -> 'CompositeBloq':
     def adjoint(self) -> 'GlobalPhase':
         return attrs.evolve(self, coefficient=sconj(self.coefficient))
 
+    def my_tensors(
+        self, incoming: Dict[str, 'ConnectionT'], outgoing: Dict[str, 'ConnectionT']
+    ) -> List['qtn.Tensor']:
+        import quimb.tensor as qtn
+
+        return [qtn.Tensor(data=self.coefficient, inds=[], tags=[str(self)])]
+
+    def get_ctrl_system(
+        self, ctrl_spec: Optional['CtrlSpec'] = None
+    ) -> Tuple['Bloq', 'AddControlledT']:
+
+        # Default logic for more than one control.
+        if not (ctrl_spec is None or ctrl_spec == CtrlSpec()):
+            return super().get_ctrl_system(ctrl_spec=ctrl_spec)
+
+        exponent: float = np.real_if_close(np.log(self.coefficient) / (np.pi * 1.0j)).item()  # type: ignore
```

Comment:
`self.coefficient` can be symbolic, in which case this line would throw an error instead of returning a symbolic `ZPowGate`. 

While this is a useful change, Why do we need to make this change as part of this PR? Is it because of the cirq-interop changes in the `Controlled` bloq? Can we pull these out in a separate PR?

Reply:
The new `Controlled.my_tensors` method does not work with GlobalPhase since its "tensor" is just a number (with the rest of the cirq changes reverted)

Reply:
This is because I'm using `subbloq.tensor_contract()` which returns a number instead of an array. Since this is the public-facing function, I don't want to change its behavior. Instead, I've put a check in `Controlled.my_tensors` to handle the case where the subbloq's tensor is just a number

Reply:
reverted and did the special case described above